cedar-policy-core/src/parser.rs

```diff
@@ -469,52 +471,51 @@ mod eval_tests {
         let entities = eval::test::basic_entities();
         let exts = Extensions::none();
         let evaluator = eval::Evaluator::new(request, &entities, &exts);
+        // The below tests check not only that we get the expected `Value`, but
+        // that it has the expected source location.
+        // We have to check that separately because the `PartialEq` and `Eq`
+        // impls for `Value` do not compare source locations.
```

Comment:
Should they? At least personally I'd expect this.

Reply:
I feel that the evaluator should not consider source locations when comparing values for equality.  Source locations are not part of the Cedar semantics.  So I think we definitely need two senses of equality for Values -- one that ignores source locations, and is used by the evaluator; and another that considers source locations, and is used in tests / for debugging / maybe other purposes.

The question here is then, which sense should get the actual `PartialEq` / `Eq` implementation, and which one should be a different method on `Value`.  I sense that you're proposing we have `PartialEq` / `Eq` be the sense of equality that considers source locations.  I'm inclined to agree.  Did I interpret correctly?

Reply:
Yeah, definitely need both and my inclination was to have `PartialEq` / `Eq` consider location. Looking at the changes needed for that I'm not as inclined now. My reasoning was that since we need both we should just use the default `derive` where we can, but I think I actually prefer the current version.

Reply:
All right, I'll leave it as is then.